internal/civisibility/integrations/gotesting/reflections.go

```diff
@@ -7,25 +7,86 @@ package gotesting
 
 import (
 	"errors"
+	"io"
 	"reflect"
 	"sync"
+	"sync/atomic"
 	"testing"
+	"time"
 	"unsafe"
 )
 
 // getFieldPointerFrom gets an unsafe.Pointer (gc-safe type of pointer) to a struct field
 // useful to get or set values to private field
 func getFieldPointerFrom(value any, fieldName string) (unsafe.Pointer, error) {
-	indirectValue := reflect.Indirect(reflect.ValueOf(value))
-	member := indirectValue.FieldByName(fieldName)
+	return getFieldPointerFromValue(reflect.Indirect(reflect.ValueOf(value)), fieldName)
+}
+
+// getFieldPointerFromValue gets an unsafe.Pointer (gc-safe type of pointer) to a struct field
+// useful to get or set values to private field
+func getFieldPointerFromValue(value reflect.Value, fieldName string) (unsafe.Pointer, error) {
+	member := value.FieldByName(fieldName)
 	if member.IsValid() {
 		return unsafe.Pointer(member.UnsafeAddr()), nil
 	}
 
 	return unsafe.Pointer(nil), errors.New("member is invalid")
 }
 
+// copyFieldUsingPointers copies a private field value from one struct to another of the same type
+func copyFieldUsingPointers[V any](source any, target any, fieldName string) error {
+	if sourcePtr, err := getFieldPointerFrom(source, fieldName); err == nil {
+		if targetPtr, err := getFieldPointerFrom(target, fieldName); err == nil {
+			*(*V)(targetPtr) = *(*V)(sourcePtr)
+		} else {
+			return err
+		}
+	} else {
+		return err
+	}
+	return nil
+}
```

Comment:
small nit
```suggestion
        sourcePtr, err := getFieldPointerFrom(source, fieldName)
	if err != nil {
	        return err
	}
	targetPtr, err := getFieldPointerFrom(target, fieldName)
	if err != nil {
		return err
	}
	*(*V)(targetPtr) = *(*V)(sourcePtr)
	return nil
}
```

Reply:
fixed in [de7d8da](https://github.com/DataDog/dd-trace-go/pull/2892/commits/de7d8da10473f370b871a2ee89b9d23aeb085f31)